### PR TITLE
[SCAL-77544]: Add copy button on code blocks

### PR DESCRIPTION
**Pull request by harsh-ts**
Ticket link -> https://thoughtspot.atlassian.net/browse/SCAL-77701, https://thoughtspot.atlassian.net/browse/SCAL-77544

This PR contains:

- Copy button displayed on code blocks.
- Copy to clipboard functionality.
- Tooltip display on copy button click.
- Code highlighting.

Here is a demo of changes:
https://user-images.githubusercontent.com/78337774/109261235-8d9e0a80-7825-11eb-8658-eeedb6c2927c.mp4

### CODE REVIEW

File: docs/src/assets/styles/highlight.scss [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    .hljs-type,
+    .hljs-class .hljs-title {
+        color: #458;
+        font-weight: $font-weight-bold;
+    }
+
+    .hljs-tag,
+    .hljs-name,
+    .hljs-attribute {
+        color: #000080;
```
Comment (pankajp-ts) [inline docs/src/assets/styles/highlight.scss:48]: color should come from a common file i.e. variable.scss

File: docs/src/components/Document/index.scss [hunk 2]
```diff
@@ -1,4 +1,5 @@
 @import '../../assets/styles/variables.scss';
+@import '../../assets//styles/highlight.scss';
```
Comment (pankajp-ts) [inline docs/src/components/Document/index.scss:2]: `assets//styles` remove extra `/`

**Review by pankajp-ts — CHANGES_REQUESTED** [review]

File: docs/src/assets/styles/highlight.scss (hunk 1 above)
Comment (harsh-ts) [reply docs/src/assets/styles/highlight.scss:48]: Okay I'll do that.

File: docs/src/components/Document/index.scss (hunk 2 above)
Comment (harsh-ts) [reply docs/src/components/Document/index.scss:2]: I'll fix that.

File: docs/src/assets/svg/copy.svg [hunk 3]
```diff
@@ -0,0 +1,48 @@
+<?xml version="1.0" encoding="iso-8859-1"?>
+<!-- Generator: Adobe Illustrator 19.0.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
```
Comment (ayonghosh) [inline docs/src/assets/svg/copy.svg:2]: Can you run this through https://jakearchibald.github.io/svgomg/ so we get rid of redundant stuff

File: docs/src/components/Document/helper.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import hljs from "highlight.js"
+import Copy from '../../assets/svg/copy.svg';
+
+export const addEventListner = (element: HTMLElement, ...args: HTMLElement[]) => {
```
Comment (ayonghosh) [inline docs/src/components/Document/helper.ts:4]: Typo in Listener

File: docs/src/components/Document/helper.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
+import hljs from "highlight.js"
```
Comment (ayonghosh) [inline docs/src/components/Document/helper.ts:1]: Use single quotes for consistency

File: docs/src/components/Document/helper.ts [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+import Copy from '../../assets/svg/copy.svg';
+
+export const addEventListner = (element: HTMLElement, ...args: HTMLElement[]) => {
+    element.addEventListener('click', () => {
+        const textareaElement = document.createElement("textarea");
+        textareaElement.value = (args[0] as HTMLElement).innerText;
+        element.parentElement.appendChild(textareaElement);
+        textareaElement.select();
+        document.execCommand("copy");
```
Comment (ayonghosh) [inline docs/src/components/Document/helper.ts:10]: Use single quotes

File: docs/src/components/Document/helper.ts [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+        spanElement.innerText = tag.getAttribute('data-lang');
+        spanElement.classList.add('lang');
+        const wrapperDiv = document.createElement('div');
+        wrapperDiv.classList.add('wrapperContainer');
+        wrapperDiv.appendChild(spanElement);
+        wrapperDiv.appendChild(buttonElement);
+        tag.parentElement.appendChild(wrapperDiv);
+    });
+    /* To highlight code snippets */
+    document.querySelectorAll("pre code").forEach(block => {
```
Comment (ayonghosh) [inline docs/src/components/Document/helper.ts:47]: Use single quotes

File: docs/src/components/Document/helper.ts [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+        divElement.appendChild(spanElement);
+        element.parentElement.appendChild(divElement);
+        /* To remove copy tooltip */
+        setTimeout(() => {
+            element.parentElement.removeChild(divElement);
+        }, 500);
+    })
+};
+
+export const customizeDocContent = () => {
```
Comment (ayonghosh) [inline docs/src/components/Document/helper.ts:26]: Possible to use event delegation here?

File: docs/src/assets/svg/copy.svg (hunk 3 above)
Comment (harsh-ts) [reply docs/src/assets/svg/copy.svg:2]: Sure. Looks useful!

File: docs/src/components/Document/helper.ts (hunk 4 above)
Comment (harsh-ts) [reply docs/src/components/Document/helper.ts:4]: Yeah. Thanks for pointing this out.

File: docs/src/components/Document/helper.ts (hunk 6 above)
Comment (harsh-ts) [reply docs/src/components/Document/helper.ts:10]: Okay

File: docs/src/components/Document/helper.ts (hunk 7 above)
Comment (harsh-ts) [reply docs/src/components/Document/helper.ts:47]: Yeah sure

File: docs/src/components/Document/helper.ts (hunk 8 above)
Comment (harsh-ts) [reply docs/src/components/Document/helper.ts:26]: Using event delegation will be expensive.

File: docs/src/components/Document/helper.ts (hunk 5 above)
Comment (harsh-ts) [reply docs/src/components/Document/helper.ts:1]: Yes it is a npm module but it's used this way.

**Review by ayonghosh — APPROVED** [review]

**Pull request merged**